src/psyclone/psyGen.py

```diff
@@ -542,41 +543,49 @@ def schedule(self):
     def schedule(self, obj):
         self._schedule = obj
 
-    def unique_declarations(self, datatype, access=None):
-        ''' Returns a list of all required declarations for the
-        specified datatype. If access is supplied (e.g. "write") then
+    def unique_declarations(self, argument_type, access=None):
+        '''
+        Returns a list of all required declarations for the specified
+        API argument type. If access is supplied (e.g. "write") then
         only declarations with that access are returned.
-        :param string datatype: The type of the kernel argument for the \
-                                particular API for which the intent is \
-                                required
-        :param access: Optional AccessType that the declaration should have.
-        :returns: List of all declared names.
-        :rtype: A list of strings.
-        :raises: GenerationError if an invalid datatype is given.
-        :raises: InternalError if an invalid access is specified.
-        '''
-        if datatype not in VALID_ARG_TYPE_NAMES:
+
+        :param string argument_type: the type of the kernel argument for \
+                                     the particular API for which the intent \
+                                     is required.
+        :param access: optional AccessType that the declaration should have.
```

Comment:
Again, not your fault but please add a `:type access:` entry.

Reply:
Heh, I knew I forgot something when updating docstrings. Added.